fastify.js

```diff
@@ -612,7 +613,12 @@ function build (options) {
 
   function setNotFoundHandler (opts, handler) {
     this.after((notHandledErr, done) => {
-      _setNotFoundHandler.call(this, opts, handler)
+      try {
+        _setNotFoundHandler.call(this, opts, handler)
+      } catch (err) {
+        done(err)
```

Comment:
I'm not sure if this is the right thing to do if `notHandledErr` is also an error.

Reply:
I don’t understand why you would want to do this.

Reply:
Can you answer this as well?

Reply:
Since this happens inside an avvio `after` callback, this is asynchronous. This means we can either throw an async error, or pass the error along to the `fastify.listen` callback. I thought passing the error forward would be better (and it's easier to test).

https://github.com/fastify/fastify/blob/e252a5c26c6d34dde1458fb5cd92c1c4656cd99d/test/404s.test.js#L146-L149

Reply:
I'm not a fan of generic try-catch blocks, as they catch _everything_. Can you move the check here?

Reply:
Actually, I think I can move the check outside of the callback and throw without a try-catch, which would provide a stack trace with the exact location of the bad call to `setNotFoundHandler`.

Reply:
Done 👍